src/frontend-samples/issues-sample/IssuesWidget.tsx

```diff
@@ -0,0 +1,504 @@
+        const targetPoint = eyePoint.plus(directionVector);
+        view3d.lookAtUsingLensAngle(eyePoint, targetPoint, upVector, Angle.createRadians(fov));
+        viewport.synchWithView();
+      }
+    }
+  }, [viewport]);
+
+  /** Create the issue marker icon, then add the pin at the issue location */
+  useEffect(() => {
+    const parser = new DOMParser();
+    const SVGMAP: { [key: string]: HTMLImageElement } = {};
+    const issue_marker: string = `
+    <svg viewBox="0 0 32 32" width="40" height="40" xmlns="http://www.w3.org/2000/svg"><path d="m25 0h-18a5 5 0 0 0 -5 5v18a5 5 0 0 0 5 5h5v.00177l4 3.99823 4-3.99823v-.00177h5a5 5 0 0 0 5-5v-18a5 5 0 0 0 -5-5z" fill="#fff" fill-rule="evenodd"/>
+      <path id="fill" d="m25 1a4.00453 4.00453 0 0 1 4 4v18a4.00453 4.00453 0 0 1 -4 4h-18a4.00453 4.00453 0 0 1 -4-4v-18a4.00453 4.00453 0 0 1 4-4z" fill="#008be1"/>
+      <path id="icon" d="m10.8125 5h1.125v18h-1.125zm12.375 6.75h-10.125v-6.75h10.125l-4.5 3.375z" fill="#fff"/>
+    </svg>`;
+
+    for (const issue of issues) {
+      const fillColor = issue.properties._StatusColor ?? "red";
+      let svg = SVGMAP[fillColor];
+      if (!svg) {
+        const imgXml = parser.parseFromString(issue_marker, "application/xml");
+
+        /** set the background fill color */
+        const fill = imgXml.getElementById("fill");
+        if (fill) {
+          fill.setAttribute("fill", fillColor);
+        }
+
+        /** set the foreground (icon flag) color */
+        const icon = imgXml.getElementById("icon");
+        if (icon) {
+          const textColor = buildForegroundColor(fillColor);
+          if (textColor) {
+            icon.setAttribute("fill", textColor);
+          }
+        }
+
+        const svgString = new XMLSerializer().serializeToString(imgXml);
+        const blob = new Blob([svgString], { type: "image/svg+xml;charset=utf-8" });
+        svg = new Image(40, 40);
+        svg.src = URL.createObjectURL(blob);
+        SVGMAP[fillColor] = svg;
+      }
+
+      /** Add the point to the decorator */
+      IssuesApi.addDecoratorPoint(issue, svg, issue.number, issue.subject, (iss: any) => {
+        applyView(iss);
+        setActiveTab(0);
+        setCurrentIssue(iss);
+      });
+    }
+  }, [applyView, issues]);
+
+  /** Helper to determine text color on the basis of background hex color. */
+  const buildForegroundColor = (markerFillColor: string): string | undefined => {
+    if (!markerFillColor) return;
+    if (markerFillColor[0] === "#") {
+      markerFillColor = markerFillColor.slice(1);
+    }
+    const r = parseInt(markerFillColor.substr(0, 2), 16);
+    const g = parseInt(markerFillColor.substr(2, 2), 16);
+    const b = parseInt(markerFillColor.substr(4, 2), 16);
+    const yiq = (r * 299 + g * 587 + b * 114) / 1000;
+    // All focus on yellow only, when user will start picking their own color, we may pick another threshhold value or change algo :|
```

Comment:
```suggestion
    // All focus on yellow only, when user will start picking their own color, we may pick another threshold value or change algo :|
```

Reply:
Actually, can you clarify? What is `yiq`. When will user pick their own color?

Reply:
This code was copy and pasted to color in the background of the SVG. The comment suggest that they could make their own colors eventually. Removed the comment